### DISCUSSION TITLE
root package install.

### DISCUSSION THREAD

**Discussion by terabytesoftw** [composer/composer#11428] [Q&A / Support]
Hi, first of all thanks for all the work done, it's awesome.

My problem:

Package A: Tag 1.0.0.

```json
{
    "name": "stub/package-a",
    "require": {
    }
}
```

Package B: Tag 1.0.0.

```json
{
    "name": "stub/package-b",
    "require": {
        "stub/package-a": "^1.0"
    }
}
```

For the tests in ci, i need install `dev-master`, when i had not released, the problem was solved by `COMPOSER_ROOT_VERSION=dev-master`, how can i solve it now.

```shell
Problem 1
    - Root composer.json requires stub/package-b dev-master -> satisfiable by stub/package-b[dev-master].
    - stub/package-b dev-master requires stub/package-a ^1.0 -> satisfiable by stub/package-a[1.0.0] from composer repo 
      (https://repo.packagist.org/) but stub/package-a is the root package and cannot be modified.
      See https://getcomposer.org/dep-on-root for details and assistance.

**Comment by stof** [comment]
I don't understand your problem. The error message seems to imply that `stub/package-a` is the root package but also that it requires `stub/package-b`. This is not possible with the package definitions you provide

**Reply by terabytesoftw** [reply]
Thanks for your answer, i will try to explain, package b is the implementation of package a, and for testing package a, i need to tests the updated master of both packages, package b is installed in package a as a dev dependency, to test the changes.

What would be the best way to implement it?

**Reply by terabytesoftw** [reply]
solve, thks.
